### ISSUE TITLE
Is this Svelte fronted still planned?

### ISSUE THREAD

**Issue by a0a7** [speechanddebate/schemats#1]
Hi, I’m wondering if the tabroom frontend is still planned to be migrated to Svelte. I ask, because I’d love to contribute code if that is the case.
Please let me know, and thank you for the incredible amount of work you’ve done for us debaters!

**Comment by vtempest** [comment]
Super happy to hear and hope to integrate this Svelte Tabroom API with a Live-Round Debate Format where all ongoing round statuses are seen. 

Here's my equivalent of Next.js in svelte https://github.com/vtempest/Serverless-DOCS-Stack

**Comment by a0a7** [comment]
I’m a bit confused sorry 😅 do you think you could rephrase/elaborate what you’re saying? Thanks :)

**Comment by palmertab** [comment]
Yup, making the plumbing of a new frontend is most of what I'll be doing in
the next few months.

Tabroom is an enormous codebase so it'll take some time, but we're
committed to at least getting to the point where no new features are
written in the legacy code.

**Comment by a0a7** [comment]
Awesome! That makes me really happy to hear. Once you _really_ get going I # would love to contribute where I can. I'm a really big fan of tabroom (well, debate, but super thankful for tabroom!) and a really big fan of svelte. Super happy that's what you're using. 

Is your plan to use svelte 4 or 5?

**Comment by FloridaMan7588** [comment]
I am also interested in contributing to this project, it would be helpful to get this running in a local development environment but in the current state of this project I can't seem to interface with the API properly. Anyways, glad to see this being done either way as a debater, this will certainly increase the usability and possibly scalability of Tabroom.

**Comment by palmertab** [comment]
yeah right now the "API" is by no means a complete thing, it's much more "Some random functions that I was able to move to node.js without affecting the current frontend in bad ways."  Porting a project incrementally is rather difficult. 

I'm making some decent headway with the Svelte framework.  I decided to pull a Leroy Jenkins and go for Svelte 5 because nothing of what I wanted to use precluded it and the energy of the community seems already centered there.  I am hoping to have some quiet time over the winter break to turn the corner and have at least the framework of the syntactic sugar, some docs and the frame of a Tabroom CSS layout/page in place so that I can start spitting out features, and also accept the same from others if they are interested. 

I will confess that collaborating with others and accepting code contributions has never been a strength of mine; I tend to focus pretty hard on what I have to do and put email/comms on the back burner a bit too much.  So don't feel it's personal if you put in a PR and I just let it sit forever; sometimes it means I'm too wrapped up in something else, and sometimes it means I've put it into the category of "good idea but I can't test it fully right now."

Here's hoping I can also get a full testing framework going too, but boy howdy, the full todo-list is daunting.  Twenty some years of code doesn't get replaced in a day, alas.

**Comment by a0a7** [comment]
Absolutely fair; sounds like good progress!

**Comment by FloridaMan7588** [comment]
I totally get it! I'm a very OSS minded person but I rarely check my projects for PRs because I just want to work on what I want to work on, and not a lot of people rely on what I do. Of course this is different because a lot of people rely on Tabroom, but I can imagine this port is a nightmare so you should take your time. I plan on contributing at some point, and totally understand if you want to prioritize other things over PRs. Best of luck on this massive port to Svelte!

**Comment by a0a7** [comment]
Hi @palmertab , sorry to take your time again. I saw that you posted a developer role to linkedin: and while I'm not in a position to be hired in a role like that, I was wondering if that might signal a little more of a collaborative dev environment for tab. 

To be entirely frank, I wanted to ask if you would be open to offering some kind of unpaid, low-key internship role (and pitch myself for a role like that).

I know directing/working with others sometimes takes a lot of time over just doing it yourself, but I think I could do a good job of managing my own tasks and being an asset rather than a burden. I do have quite substantial experience with fullstack Svelte development, and familiarity with sql, node, linux and other tools I know you make use of. 
If you're open to it please do let me know and I would be grateful for the opportunity to elaborate & set up a time to talk. If not I completely understand :)
Either way, I appreciate all you do for us debaters.
Thanks,
Alexander
